File: app/src/main/java/com/musicplayer/SocyMusic/ui/playlists_tab/PlaylistHolder.java
package com.musicplayer.SocyMusic.ui.playlists_tab;

import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.musicplayer.SocyMusic.data.Playlist;
import com.musicplayer.musicplayer.R;

public class PlaylistHolder extends RecyclerView.ViewHolder {
    private final TextView playlistTitleTextView;
    private final TextView songCountTextView;
    private final PlaylistsAdapter.ItemClickListener clickListener;

    public PlaylistHolder(@NonNull View itemView, PlaylistsAdapter.ItemClickListener clickListeners) {
        super(itemView);
        playlistTitleTextView = itemView.findViewById(R.id.textview_playlist_item_name);
        songCountTextView = itemView.findViewById(R.id.textview_playlist_item_songcount);
        this.clickListener = clickListeners;
        itemView.setOnClickListener(v -> clickListeners.onItemClick(getBindingAdapterPosition(), v));
        itemView.setOnLongClickListener(v -> clickListeners.onItemLongClick(getBindingAdapterPosition(), v));
    }

    public void bind(Playlist playlist) {
        playlistTitleTextView.setText(playlist.getName());
        songCountTextView.setText(itemView.getContext().getString(R.string.playlist_song_count, playlist.getSongCount()));
    }

}

File: app/src/main/kotlin/com/musicplayer/openmusic/ui/playlists_tab/PlaylistHolder.kt
package com.musicplayer.openmusic.ui.playlists_tab

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.musicplayer.openmusic.data.Playlist
import com.musicplayer.musicplayer.R

class PlaylistHolder(itemView: View, clickListeners: PlaylistsAdapter.ItemClickListener) :
    RecyclerView.ViewHolder(itemView) {
    private val playlistTitleTextView: TextView
    private val songCountTextView: TextView
    private val clickListener: PlaylistsAdapter.ItemClickListener

    init {
        playlistTitleTextView = itemView.findViewById(R.id.textview_playlist_item_name)
        songCountTextView = itemView.findViewById(R.id.textview_playlist_item_songcount)
        clickListener = clickListeners
        itemView.setOnClickListener { v: View ->
            clickListeners.onItemClick(
                bindingAdapterPosition, v
            )
        }
        itemView.setOnLongClickListener { v: View ->
            clickListeners.onItemLongClick(
                bindingAdapterPosition, v
            )
        }
    }

    fun bind(playlist: Playlist) {
        playlistTitleTextView.text = playlist.name
        songCountTextView.text =
            itemView.context.getString(R.string.all_song_count, playlist.songCount)
    }
}